jpa/subsystem/src/main/java/org/jboss/as/jpa/subsystem/JPADefinition.java

```diff
@@ -72,25 +74,41 @@ public class JPADefinition extends SimpleResourceDefinition {
 
     JPADefinition(boolean feature) {
         super(getParameters(feature));
+        this.deployment = !feature;
     }
 
     private static Parameters getParameters(boolean feature) {
         Parameters result = new Parameters(PathElement.pathElement(ModelDescriptionConstants.SUBSYSTEM, JPAExtension.SUBSYSTEM_NAME),
                 JPAExtension.getResourceDescriptionResolver())
                 .setFeature(feature).setRuntime(!feature);
         if (feature) {
-            result = result.setCapabilities(JPA_CAPABILITY);
+            result.setCapabilities(JPA_CAPABILITY)
+                    .setAddHandler(JPASubSystemAdd.INSTANCE)
+                    .setRemoveHandler(ReloadRequiredRemoveStepHandler.INSTANCE);
         }
-        // TODO WFLY-11173 add/remove handlers don't make sense on the deployment resource
-        result = result.setAddHandler(JPASubSystemAdd.INSTANCE).setRemoveHandler(ReloadRequiredRemoveStepHandler.INSTANCE);
-
         return result;
     }
 
     @Override
     public void registerAttributes(ManagementResourceRegistration registration) {
-        registration.registerReadWriteAttribute(DEFAULT_DATASOURCE, null, new ReloadRequiredWriteAttributeHandler(DEFAULT_DATASOURCE));
-        registration.registerReadWriteAttribute(DEFAULT_EXTENDEDPERSISTENCE_INHERITANCE, null, new ReloadRequiredWriteAttributeHandler(DEFAULT_EXTENDEDPERSISTENCE_INHERITANCE));
+        // WFLY-11173
+        // In case of /deployment=* the attributes cannot be written, only read. If read, the main /subsystem=jpa values are returned.
+        if (deployment) {
+            registration.registerReadOnlyAttribute(DEFAULT_DATASOURCE, (context, operation) -> {
+                PathElement subsystemAddress = PathElement.pathElement(ModelDescriptionConstants.SUBSYSTEM, JPAExtension.SUBSYSTEM_NAME);
+                ModelNode modelNode = context.getOriginalRootResource().getChild(subsystemAddress).getModel().get(DEFAULT_DATASOURCE.getName());
```

Comment:
This should be

context.readResourceFromRoot(subsystemAddress, false)

The only place where this matters is if this attribute was read as a step in a composite op that wrote the subsystem attribute. When that happens, the read should reflect the change made by the write. 

The getOriginalRootResource() is pretty much just for cases where an OSH needs to modify runtime services and as part of doing that it needs to know the original value of something. For example as part of rolling back a change in an op that failed.